c10/core/InferenceMode.h

```diff
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <ATen/core/grad_mode.h>
```

Comment:
IMO, this `include` breaks library hierarchy, where c10 is foundatinal layer and Aten is build on top of it.
Could `grad_mode.h` be moved into c10 folder and namespace?